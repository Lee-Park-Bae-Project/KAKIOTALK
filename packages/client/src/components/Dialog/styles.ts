import styled from 'styled-components'
import Flex from 'atoms/Flex'
import {
  color, fontSize, weight,
} from 'styles/global'

export const Container = styled(Flex)`

`

export const ContentWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`

export const Content = styled(Flex)`
  flex-direction: column;
  
  background-color: white;
  width: 20rem;
  padding: 2rem;
`

export const Title = styled.p`
  font-size: ${fontSize.LARGE};
  color: ${color.BLACK};
  font-weight: ${weight.STRONG};
  margin-right: auto;
`

export const Description = styled.p`
  font-size: ${fontSize.MEDIUM};
  color: ${color.BLACK};
  font-weight: ${weight.NORMAL};
  margin-right: auto;
`
