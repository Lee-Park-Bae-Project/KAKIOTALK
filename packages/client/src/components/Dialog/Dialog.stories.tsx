import React, { useState } from 'react'
import Dialog from 'components/Dialog'
import {
  boolean, text, withKnobs,
} from '@storybook/addon-knobs'
import UserCard from 'components/UserCard'
import Button from 'components/Button'
import { color } from 'styles/global'

export default {
  title: 'Component/Dialog',
  component: Dialog,
  parameters: { docs: { inlineStories: false } },
  decorators: [withKnobs],
}

export const DefaultDialog = () => {
  const isVisible = boolean('isVisible', true)
  const title = text('title', '친구추가')
  const description = text('description', 'junow 님을 친구목록에 추가하시겠습니까?')
  const isHideButton = boolean('isHideButton', false)
  const canCancel = boolean('canCancel', true)
  const cancelText = text('cancelText', '취소')
  const confirmText = text('confirmText', '확인')

  return (
    <Dialog
      isVisible={isVisible}
      title={title}
      description={description}
      isHideButton={isHideButton}
      canCancel={canCancel}
      cancelText={cancelText}
      confirmText={confirmText}
    />
  )
}

export const WithChildren = () => {
  const isVisible = boolean('isVisible', true)
  const isHideButton = boolean('isHideButton', true)
  const canCancel = boolean('canCancel', true)
  const cancelText = text('cancelText', '취소')
  const confirmText = text('confirmText', '확인')
  const name1 = text('name1', 'junow')
  const statusMessage1 = text('statusMesage1', 'I Like Juno')
  const name2 = text('name2', 'marshmello3773')
  const statusMessage2 = text('statusMessage2', 'I Like Marshmello')
  const name3 = text('name3', 'Baebaam')
  const statusMessage3 = text('statusMessage3', 'I Like Travel')

  return (
    <Dialog
      isVisible={isVisible}
      isHideButton={isHideButton}
      canCancel={canCancel}
      cancelText={cancelText}
      confirmText={confirmText}
    >
      <UserCard
        name={name1}
        statusMessage={statusMessage1}
      />
      <UserCard
        name={name2}
        statusMessage={statusMessage2}
      />
      <UserCard
        name={name3}
        statusMessage={statusMessage3}
      />
    </Dialog>
  )
}

export const WithClickListener = () => {
  const [visible, setVisible] = useState(false)
  const [result, setResult] = useState('')
  const title = text('title', '버튼 클릭 테스트')
  const description = text('description', '버튼 클릭 테스트')
  const isHideButton = boolean('isHideButton', false)
  const canCancel = boolean('canCancel', true)
  const cancelText = text('cancelText', '취소')
  const confirmText = text('confirmText', '확인')

  const onClick = () => {
    setVisible(true)
  }

  const onCancel = () => {
    setVisible(false)
    setResult('취소함')
  }

  const onConfirm = () => {
    setVisible(false)
    setResult('확인함')
  }
  return (
    <>
      <Button
        text='dialog 열기'
        onClick={onClick}
      />
      <p style={{ color: color.YELLO }}>{result}</p>
      <Dialog
        isVisible={visible}
        title={title}
        description={description}
        isHideButton={isHideButton}
        canCancel={canCancel}
        cancelText={cancelText}
        confirmText={confirmText}
        onCancel={onCancel}
        onConfirm={onConfirm}
      />
    </>
  )
}
