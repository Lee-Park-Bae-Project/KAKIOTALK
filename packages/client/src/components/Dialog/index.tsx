import React, { FC } from 'react'
import * as S from 'components/Dialog/styles'
import Button from 'components/Button'
import ButtonGroup from 'components/ButtonGroup'

interface Props{
  /** 렌더링 여부 */
  isVisible: boolean;
  /** 제목 */
  title?: string;
  /** 설명 */
  description?: string;
  /** 하위요소들 */
  children?: React.ReactNode;
  /** 기본 버튼 보일지 안보일지 */
  isHideButton: boolean;
  /** 단순 취소가 가능한지 */
  canCancel: boolean;
  /** 취소 버튼에 들어갈 글자 */
  cancelText: string;
  /** 확인 버튼에 들어갈 글자 */
  confirmText: string;
  /** 취소버튼 클릭 핸들러 */
  onCancel?: () => void;
  /** 확인 버튼 클릭 핸들러 */
  onConfirm?: () => void;
}

/**
 * 다이얼로그 입니다.
 * 기본적으로는 제목과 설명글로 렌더링가능하고
 * 필요한 경우 하위요소를 넘겨서 렌더링 할 수 있습니다.
 */
const Dialog: FC<Props> = ({
  isVisible,
  title,
  description,
  children,
  isHideButton,
  canCancel,
  cancelText,
  confirmText,
  onCancel,
  onConfirm,
}) => {
  if (!isVisible) {
    return null
  }
  return (
  <S.Container>
    <S.ContentWrapper>
      <S.Content>
        {title && <S.Title>{title}</S.Title>}
        {description && <S.Description>{description}</S.Description>}
        {children}
        {!isHideButton && (
          <ButtonGroup right>
            {canCancel && (
              <Button
                text={confirmText}
                theme='primary'
                onClick={onConfirm}
              />
            )}
            <Button
              text={cancelText}
              theme='tertiary'
              onClick={onCancel}
            />
          </ButtonGroup>
        )}
      </S.Content>
    </S.ContentWrapper>
  </S.Container>
  )
}

export default Dialog
