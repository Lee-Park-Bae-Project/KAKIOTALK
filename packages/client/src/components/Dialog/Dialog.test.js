/* eslint-disable no-undef */
import React from 'react';
import { mount } from 'enzyme';
import Dialog from 'components/Dialog';

describe('<Dialog/>', () => {
  it('matches snapshot', () => {
    const isVisible = true;
    const title = 'this is title';
    const description = 'this is description';
    const isHideButton = false;
    const canCancel = true;
    const cancelText = 'cancel';
    const confirmText = 'confirm';

    const wrapper = mount((
      <Dialog
        isVisible={isVisible}
        title={title}
        description={description}
        isHideButton={isHideButton}
        canCancel={canCancel}
        cancelText={cancelText}
        confirmText={confirmText}
      />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('matches props', () => {
    const isVisible = true;
    const title = 'this is title';
    const description = 'this is description';
    const isHideButton = false;
    const canCancel = true;
    const cancelText = 'cancel';
    const confirmText = 'confirm';

    const wrapper = mount((
      <Dialog
        isVisible={isVisible}
        title={title}
        description={description}
        isHideButton={isHideButton}
        canCancel={canCancel}
        cancelText={cancelText}
        confirmText={confirmText}
        id="dialog"
      />
    ));

    expect(wrapper.props().isVisible).toEqual(true);
    expect(wrapper.props().title).toEqual(title);
    expect(wrapper.props().description).toEqual(description);
    expect(wrapper.props().isHideButton).toEqual(isHideButton);
    expect(wrapper.props().canCancel).toEqual(canCancel);
    expect(wrapper.props().cancelText).toEqual(cancelText);
    expect(wrapper.props().confirmText).toEqual(confirmText);
  });

  it('event fire (just change text)', () => {
    let result = '';
    const isVisible = true;
    const title = 'this is title';
    const description = 'this is description';
    const isHideButton = false;
    const canCancel = true;
    const cancelText = 'cancel';
    const confirmText = 'confirm';

    const cancelResult = 'cancel';
    const confirmResult = 'confirm';

    const onCancel = () => {
      result = cancelResult;
    };
    const onConfirm = () => {
      result = confirmResult;
    };

    const wrapper = mount((
      <Dialog
        isVisible={isVisible}
        title={title}
        description={description}
        isHideButton={isHideButton}
        canCancel={canCancel}
        cancelText={cancelText}
        confirmText={confirmText}
        id="dialog"
        onCancel={onCancel}
        onConfirm={onConfirm}
      />
    ));

    wrapper.find('button').at(0).simulate('click');
    expect(result).toEqual(confirmResult);
    wrapper.find('button').at(1).simulate('click');
    expect(result).toEqual(cancelResult);
  });
});
