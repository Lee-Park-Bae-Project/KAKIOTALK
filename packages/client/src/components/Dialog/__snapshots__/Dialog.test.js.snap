// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog/> matches snapshot 1`] = `ReactWrapper {}`;
